manifests/tunnel.pp

```diff
@@ -3,31 +3,49 @@
   $source,
   $dest,
   $secret,
+  $user
 ) {
   ensure_resource('package', 'spiped', {'ensure' => 'present'})
   ensure_resource('file', '/etc/spiped', {'ensure' => 'directory'})
 
-  $unitfile = "/lib/systemd/system/spiped-${title}.service"
   $keyfile = "/etc/spiped/${title}.key"
 
+  if $::operatingsystem == 'Ubuntu' {
+    $unitfile = "/etc/init/spiped-${title}.conf"
+    $content = template('spiped/spiped.conf.erb')
+    $command = "restart spiped-${title}"
+  }
```

Comment:
The latest Ubuntu releases use systemd too, so I think we'd need a slightly more intelligent check here. But I'm not sure off the top of my head how to write it.